src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/TimeStatement.java

```diff
@@ -191,65 +183,24 @@ public Node visitTimeQualifier(DPLParser.TimeQualifierContext ctx) {
      * @return ElementNode(XML) with LE/GE unixtime
      */
     private ElementNode timeQualifierEmitXml(DPLParser.TimeQualifierContext ctx) {
-        String op = null;
-        long timevalue = 0;
-        String value = null;
-        boolean isRelativeTime = false;
-        Token comparisonToken;
-
         // Get specifier. We know that 2 first childs are terminals
         // 'earliest = '
         TerminalNode node = (TerminalNode) ctx.getChild(0);
-        value = ctx.getChild(1).getText();
-        Timestamp now = new Timestamp(System.currentTimeMillis());
-        RelativeTimeParser rtParser = new RelativeTimeParser();
-        // Is time given as absolute
-        // Try to check if it is relative and catch exception
-        try {
-            // relative time
-            RelativeTimestamp rtTimestamp = rtParser.parse(value); // might throw NFE if not relative timestamp
-            timevalue = rtTimestamp.calculate(now);
+        String value = ctx.getChild(1).getText();
+
+        TimeQualifier tq = new TimeQualifier(value, catCtx.getTimeFormatString(), node.getSymbol().getType(), doc);
```

Comment:
i think the syntax supports specifying timeformat multiple times, here it is on the catCtx indicating only one timeformat would be supported for the whole query, however

```
index=search_bench timeformat=%s earliest=1 timeformat=%Y-%m-%dT%H:%M:%S latest=2024-11-14T08:00:00
```

should be a valid query as well.

perhaps it is a worth of another issue?